src/mqtt_client.rs

```diff
@@ -285,6 +285,16 @@ where
         Ok(())
     }
 
+    /// Finish sending last packet if the previous operation returned Error::PartialWrite
+    ///
+    /// # Note
+    /// If this function is not called after partial write until the packet is
+    /// fully sent, the packets will most likely seem malformed to the receiving
+    /// party and they will close the connection.
+    pub fn finish_write(&mut self) -> Result<(), Error<TcpStack::Error>> {
```

Comment:
Let's add a call to `client.finish_write()` on every execution of the `Minimq.poll()` function to flush these messages automatically. There's no need for the user to manually do this.